internal/services/paloalto/local_rule_stack_rule_resource.go

```diff
@@ -0,0 +1,542 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/paloalto/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type LocalRuleStackRule struct{}
+
+var _ sdk.ResourceWithUpdate = LocalRuleStackRule{}
+
+var protocolApplicationDefault = "application-default"
+
+type LocalRuleModel struct {
+	Name        string `tfschema:"name"`
+	RuleStackID string `tfschema:"rule_stack_id"`
+	Priority    int    `tfschema:"priority"`
+
+	Action                  string                 `tfschema:"action"`
+	Applications            []string               `tfschema:"applications"`
+	AuditComment            string                 `tfschema:"audit_comment"`
+	Category                []schema.Category      `tfschema:"category"`
+	DecryptionRuleType      string                 `tfschema:"decryption_rule_type"`
+	Description             string                 `tfschema:"description"`
+	Destination             []schema.Destination   `tfschema:"destination"`
+	LoggingEnabled          bool                   `tfschema:"logging_enabled"`
+	InspectionCertificateID string                 `tfschema:"inspection_certificate_id"` // This is the name of a Certificate resource belonging to the SAME LocalRuleStack as this rule
+	NegateDestination       bool                   `tfschema:"negate_destination"`
+	NegateSource            bool                   `tfschema:"negate_source"`
+	Protocol                string                 `tfschema:"protocol"`
+	ProtocolPorts           []string               `tfschema:"protocol_ports"`
+	RuleEnabled             bool                   `tfschema:"enabled"`
+	Source                  []schema.Source        `tfschema:"source"`
+	Tags                    map[string]interface{} `tfschema:"tags"`
+
+	// Computed
+	Etag string `tfschema:"etag"` // TODO - Expose this here?
+}
+
+func (r LocalRuleStackRule) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return localrules.ValidateLocalRuleID
+}
+
+func (r LocalRuleStackRule) ResourceType() string {
+	return "azurerm_palo_alto_local_rule_stack_rule"
+}
+
+func (r LocalRuleStackRule) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.LocalRuleStackRuleName, // TODO - Check this
+		},
+
+		"rule_stack_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: localrules.ValidateLocalRulestackID,
+		},
+
+		"priority": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IntBetween(1, 10000),
+		},
+
+		// Optional
+
+		"action": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Default:      string(localrules.ActionEnumAllow),
```

Comment:
since this is security related, I'd argue this should be Required/not-defaulted?